### ISSUE TITLE
[QUESTION]: Evidential regularization coefficient

### ISSUE THREAD

**Issue by msk235** [chemprop/chemprop#352]
Hi all, 

I have a question regarding the evidential regularization coefficient used for training with the evidential loss function. 

The default value in chemprop is set to `0`. However, in the `args.py` file, the comment documenting the code states that the value used in literature is `1`. In the paper Soleimany et al (2021) however, which you refer to regarding your evidential methods, they state that the default value should be `0.2`. 

I am curious why you have chosen `0` as the default value for the the evidential regularization coefficient. 
Why did you not choose `1`, as specified in the documentation or `0.2` as stated in the Soleimany paper?

Thanks for your help!

**Comment by cjmcgill** [comment]
Hi, here is the thinking that went into this.

With regards to the docstring, the implementation of Soleimany et al.(2021) can be found in https://github.com/aamini/chemprop. The default argument for regularization in that repo is 1.0, and when implementing this we referred to this value accidentally instead of the paper recommendation. The docstring will be corrected to reflect this.

It was our view that the default should be the simplest implementation of the loss functions, one without regularization. Users can add complexity in the form of regularization, but we want them to make the decision to do so rather than by default.

The optimal degree of evidential regularization is dataset dependent. If you view Figure S6 of the Soleimany paper, you can see that the optimal level of regularization is fairly different among the four small datasets tested (0.4, 1.0, 0.2, Nan). Similar testing was not carried out on the larger datasets. 

Given the narrowness of that testing of optimal regularization factor in Soleimany et al.(2021) and the fact that 0.0 was the best choice for the largest dataset tested, we didn't view the case for using default value of 0.2 as compelling for all datasets. The difference between 0.0 and 0.2 as default values in a generalized sense seems small or uncertain, which we viewed as being outweighed by the interest of default simplicity.

We think that users who want to use regularization for this will/should perform their own regularization optimization instead of relying on a default. The docstring will be updated to reflect this.

**Comment by msk235** [comment]
Hi @cjmcgill 

thank you so much for your quick reply and your detailed explanation. 

It is very helpful to get an insight view into the thinking that went into your decisions.
